lib/resolveGitRepo.js

```diff
@@ -0,0 +1,45 @@
+'use strict'
+
+const normalize = require('normalize-path')
+const path = require('path')
+
+const execGit = require('./execGit')
+const { readBufferFromFile } = require('./file')
+
+/**
+ * Resolve path to the .git directory, with special handling for
+ * submodules
+ */
+const resolveGitConfigDir = async ({ gitDir, isSubmodule }) => {
+  const defaultDir = path.resolve(gitDir, '.git')
+  if (!isSubmodule) return normalize(defaultDir)
+
+  const buffer = await readBufferFromFile(defaultDir)
+  const dotGit = buffer.toString()
+  const gitConfigDir = path.resolve(dotGit.replace(/^gitdir: /, ''))
```

Comment:
I believe it does not work properly because of this line. It takes path relative to node_modules or package.json (not sure), and from this going up a directory. I think it should be something like
`  const gitConfigDir = path.resolve(gitDir + '/' + dotGit.replace(/^gitdir: /, ''))`
but probably better written

Reply:
Thanks for catching this. I tested it myself locally and the path in `.git` seems to be relative to the `gitDir`, so that would be better indeed.

Reply:
one more thing,  in my case 
`dotGit.replace(/^gitdir: /, '')` contains new line at the end, so it tries to read from 
path/path/path
MERGE_MSG
when i add .trim() after replace it reads from proper path.

With this change and above change to path it works great in my case 🥇

Reply:
Those following newlines could use some trimming in other places as well, let me fix... thanks for testing!

Reply:
Done! Maybe a re-test just in case?

Reply:
Works great now